Repository Cleanup

ElectronicsArchiver opened the pull request:
[Badge Suggestions]: https://img.shields.io/badge/Ｒｅｐｏｓｉｔｏｒｙ_Ｓｕｇｇｅｓｔｉｏｎｓ-609926?style=for-the-badge&logoColor=white&logo=GitHub
[Badge Changes]: https://img.shields.io/badge/Ｃｈａｎｇｅｓ-D81939?style=for-the-badge&logoColor=white&logo=GitLFS
[Badge Ask]: https://img.shields.io/badge/Ｐｌｅａｓｅ_Ａｓｋ_Ｍｅ-73398D?style=for-the-badge&logoColor=white&logo=AskUbuntu


[Widgets]: https://github.com/MarkedDown/Disable/blob/main/Features/Widgets.md
[Tabs]: https://github.com/MarkedDown/Disable/blob/main/Features/Tabs.md


<br>
<br>

<div align = center>

[<kbd> <br> Please Check The **Preview** <br> </kbd>][Preview]

</div>
  
<br>
<br>

## ![Badge Suggestions]

  <br>

- **Add Description**

  <br>

- ***[Disable Unused Tabs][Tabs]***:

  - `Projects`

  - `Wiki`

- **License Clarification**
  
  *The license states `darkfrei` to be a license holder,*
  *however, in the mods description you stated that this*
  *mod is written from scratch.*

  *If you only adopted the idea of darkfrei but no content*
  *then the license shouldn't list them, rather you could*
  *add a `Acknowledgement` section.*


<br>
<br>

## ![Badge Changes]

- Moved source files into `/Source/`

  - Fixed workflow

  - Fixed zipping script

- Created basic README
 
  - Added badges:
    
    - License
    - Discord
    - Reddit

  - Added Quicklinks:
   
    - Download
    - Changelog
   
  - Added logo / thumbnail

  - Added build instructions

- Corrected license name

- Adjusted English localization

<br>
<br>

## ![Badge Ask]

- ***If you have any questions.***

- ***In case you do not agree with some of the changes made in this PR***, <br>
  *please first let me know about them, before disregarding it entirely so*<br>
  *we can possibly find a middle ground and have it not go to waste.*

<br>
<br>




[Preview]: https://github.com/ElectronicsArchiver/factorio-WhereIsMyBody

ZwerOxotnik commented:
\*Coughs\*  
Are you a bot or semi-bot?

ZwerOxotnik commented:
I'm sorry for my rudeness, you're clearly not a bot.

ZwerOxotnik commented:
I'm pretty interested in this discussion. And I actually have some templates for Factorio projects. For instance: [https://github.com/ZwerOxotnik/zk-lib](https://github.com/ZwerOxotnik/zk-lib)

You have a very important point.

> _If you only adopted the idea of darkfrei but no content_  
> _then the license shouldn't list them, rather you could_  
> _add a `Acknowledgement` section._

In short, you're right, and I have to ponder about it more. I'm not a lawyer etc., so I have lack of understanding of such stuff and I should learn about it etc. more. Also, I don't fully understand if someone made a commit and I added to a repository, then what I should about it? Well, it sounds a bit silly, but I have too many thoughts about it.

\> Moved source files into `/Source/`

It's pretty subjective in some cases, but Factorio needs `locale` folder to be in the main folder or in a scenario, also conceptually by Factorio source files in the main folder, so some files like `info.json`, `control.lua` etc. should be in the folder. And I use it in such way because I consider them in a less "monolith" way, especially with dynamic languages.  
[https://wiki.factorio.com/Tutorial:Localisation](https://wiki.factorio.com/Tutorial:Localisation)  
https://lua-api.factorio.com/latest/Data-Lifecycle.html  
   
\> ElectronicsArchiver@708acc3  
I don't fully understand why you changed the file name. It's not like I use/included a multi-license etc. but still.

\> Also, e49f00d  
I guess, we should invite someone else to clarify some stuff at some point and to be more certain about some stuff.  

I'll discuss more later, it's a bit overwhelming, please remind me if I'll forget about it. Currently, I consider that we should talk rather, ask more questions, no commits.

What you like to get from this discussion or what you want to achieve?

ElectronicsArchiver commented:
> And I actually have some templates for Factorio projects. 

Ah, I see.

>  don't fully understand if someone made a commit and I added to a repository, then what I should about it? 

A person committing -> PR -> accepted makes that contribution 
to be under the same license as the repo is.

This also means that you cannot change the license unless all
contributors that have active content in the repo aggree to
the license change.

*Though usually a license doesn't change to begin with.*

> It's pretty subjective in some cases, but Factorio needs ...

Well source would be the new 'main' folder.

Not wanting to clutter the root folder of a repo is pretty standard,
bad enough that repos have dozens of config files in the root as well.

A general structure that many repos use is ( depending on need obviously ):

- Resources / Assets ( often just for the repo, not the content)
- Source ( code, in this case mod content like local, models, .. )
- Build ( usually .gitignored )
- Documentation
- ...

> I don't fully understand why you changed the file name. It's not like I use/included a multi-license etc. but still.

The default name is just `LICENSE` there is no need to add  a `.txt` extension.

*Exception of course being licenses that have been formatted with markdown of course.*

> I guess, we should invite someone else to clarify some stuff at some point and to be more certain about some stuff.  

Corrected the description but also reworded it to be more clear.

I capitalized the - what I would assume to be - setting headlines,
as they act more as title, not descriptions and as such are more
readable if capitalized fully.

> What you like to get from this discussion or what you want to achieve?

I'm not sure what you mean, this is a PR, the goal is quite clear, is it not?

ZwerOxotnik commented:
Out of curiosity, I noticed that you have got used to have/see more space between content. Am I right?

ElectronicsArchiver commented:
I think you mean the spaces between the badges / buttons?

If so, yes, to add more horizontal spacing you can
simply use the other 6~ space types to add varying
degrees of space.

https://unicode-table.com/en/search/?q=space

ElectronicsArchiver commented:
Currently I usually use powers of 3-Per-EM 
spaces as my browser seems to have a bug
displaying wider space types.

*Of course they work for everyone else.*

*Talked to support, seems to be literally only me. x3*

ElectronicsArchiver commented:
Btw, not sure if you're interested, but I 
also have some general **[Markdown]** guides.

*At some point I got tired of explaining stuff to people~*

[Markdown]: https://github.com/MarkedDown

ZwerOxotnik commented:
Those styles are pretty new to me. I never saw such styles.

> I'm not sure what you mean, this is a PR, the goal is quite clear, is it not?

Well, I'm very surprised to see you, because it seems like you're a not a Lua developer nor a Factorio player who wants to add a new feature to this mod to play with the feature in the game. Also, it seems like your interest is, kind of, somehow outside current gaming/Factorio/Lua community and it seems like this PR is, kind of, less important etc. than this discussion for you/us.

ElectronicsArchiver commented:
> Those styles are pretty new to me. I never saw such styles.

Well depends on how much you look around I guess,
as well as what kinds of repos you look at, most
repositories have outdated docs, formatting or 
some even don't have a README ;D

> Well, I'm very surprised to see you, because it seems like you're a not a Lua developer nor a Factorio player who wants to add a new feature to this mod to play with the feature in the game. Also, it seems like your interest is, kind of, somehow outside current gaming/Factorio/Lua community and it seems like this PR is, kind of, less important etc. than this discussion for you/us.

I actually do play Factorio and do some stuff with lua, just not on this account,
here I focus ( usually ) on documentation / general structure of repositories,
though I sometimes also do code refactorings or write tools, like for [Evolvere Icons].

[Evolvere Icons]: https://github.com/franksouza183/Evolvere-Icons

ZwerOxotnik commented:
Fair enough...

Excuse me, if you consider me picky right now or something like that. (it's not important now, but still)

> The default name is just `LICENSE` there is no need to add a `.txt` extension.

If I remember correctly, the default name is `LICENSE` for new GitHub repositories, however it's fine to provide another name in some cases and other licenses provide different names. I consider `.txt` as a pretty fine choice in general. Is it possible to have not a technical issue if someone renamed some license file?

ElectronicsArchiver commented:
> Excuse me, if you consider me picky right now or something like that. (it's not important now, but still)

No worries~

> If I remember correctly, the default name is LICENSE for new GitHub repositories, however it's fine to provide another name in some cases and other licenses provide different names. I consider .txt as a pretty fine choice in general. 

Well people usually don't call it `LICENSE` when they either don't know what 
to call it or if it's an old project that still uses the `COPYING` naming.

> Is it possible to have not a technical issue if someone renamed some license file?

A non-technical issues?

ZwerOxotnik commented:
> A non-technical issues?

Some licenses don't allow to change/use their license names, but does it include file names also? I hope it's not, however, considering laws/cases etc., there are weird stuff to me that somehow legal or illegal.

> This also means that you cannot change the license unless all  
> contributors that have active content in the repo aggree to  
> the license change.

Well, ah, I should learn more. It's pretty interesting to see how it'd happen if a project has several/multi-licenses or someone has special rights.  
Also, it's MIT, so someone can copy and use a different license without any issue.

ElectronicsArchiver commented:
> Some licenses don't allow to change/use their license names, but does it include file names also? I hope it's not, however, considering laws/cases etc., there are weird stuff to me that somehow legal or illegal.

Generally speaking there shouldn't be an issue.

GitHub detects licenses by content so as long
as you haven't modifyied it, it will be detected.

Most licenses don't require anything to be inserted,
unlike the MIT license that still requires the name
and year.

For example GNU GPL, AGPL, .. licenses are simply
copyied 1 to 1 and don't require any additons.

Check out https://choosealicense.com/licenses/

> Well, ah, I should learn more. It's pretty interesting to see how it'd happen if a project has several/multi-licenses or someone has special rights.

Licenses of libraries don't incluence your codes / contents license.

No one has special rights if they contribute 
under the repositories license.

ZwerOxotnik commented:
> Moved source files into /Source/

Long story short, only that change that I should/have to deny in this case.

> Well source would be the new 'main' folder.

If I'm going to do that, then it'll be much harder to maintain the project/mod, because Factorio simply won't see the mod from git source and I'll lose the advantage of semi-live developing. (Factorio loads `/control.lua` after each restart/new game and some stuff I already mentioned in [https://github.com/ZwerOxotnik/factorio-WhereIsMyBody/pull/1#issuecomment-1162073233)](https://github.com/ZwerOxotnik/factorio-WhereIsMyBody/pull/1#issuecomment-1162073233)

\> e055ceb  
Factorio supports scenarios as .zip files and because I make all of my repositories more universal to any project. I should change that, so it'll ignore all zip files, except the ones that in `/scenarios`.

> Adjusted English localization

I guess, your variant seems more acceptable in general.  
 

I'm still thinking about the .md style etc.

Btw, I only now noticed that all of your lines has pretty "limited" amount of characters. But why here? GitHub adapts text elements to any screen size.

ZwerOxotnik commented:
[https://github.com/MarkedDown/Links/blob/main/Sections/Separators.md](https://github.com/MarkedDown/Links/blob/main/Sections/Separators.md)  
Hm, well, it seems about right in general.

ZwerOxotnik commented:
This looks off

![](https://user-images.githubusercontent.com/16271098/174987933-b5271276-50c4-435e-a024-bbc6665a64ab.png)

ElectronicsArchiver commented:
> If I'm going to do that, then it'll be much harder to maintain the project/m

Doesn't Factorio support symlinks?

That way you could just link to the source 
folder as if it was the main folder.

> Btw, I only now noticed that all of your lines has pretty "limited" amount of characters. But why here? GitHub adapts text elements to any screen size.

Most programs can in theory 'just automatically break text' 
for you, however that fails for the exact opposite reason.

They break it for small screen sizes, not for large ones.
You end up with unreadable line lengths ( > 80 ).

*Actually  > 60, human perception on average only allows*
*you to read through that many characters before you skip*
*from left to right and makes things harder to read.*

*Basicall I'm doing the work of a typesetter, there is a*
*reason why books, articles ( offline / online ),.. have*
*limited line lenghs.*

ElectronicsArchiver commented:
> This looks off
> 
> ![](https://user-images.githubusercontent.com/16271098/174987933-b5271276-50c4-435e-a024-bbc6665a64ab.png)

O.O maybe you are the first person 
with the same bug as me after all?!

Tell me, does [this page] have increasing
gaps between the block or are they mostly
the same size?

[this page]: https://github.com/MarkedDown/Whitespace

ZwerOxotnik commented:
> Doesn't Factorio support symlinks?

Any mod devs will be forced to this manual inconvenience and also it'll desync the game, so the symlink should also be removed if they want to play with the mod in multiplayer. (Factorio prefers folders over zip files)

> are they mostly  
> the same size?

![image](https://user-images.githubusercontent.com/16271098/175044567-be298468-dcc3-42bb-b780-86fdfbe5dc5e.png)

ElectronicsArchiver commented:
> Any mod devs will be forced to this manual inconvenience and also it'll desync the game, so the symlink should also be removed if they want to play with the mod in multiplayer. (Factorio prefers folders over zip files)

One symlink isn't exactly the world compared 
to many other setups one needs to do for dev.

*Plus for that a script can be used*

But you do you~

>> are they mostly
the same size?

Huh, weird, you don't have a problem with 
spaces but that example is misaligned...

ZwerOxotnik commented:
> One symlink

I often switch and when players have crushes, I have to act fast. I have a lot of mods...

ElectronicsArchiver commented:
Mhm I see, well too bad there isn't a mod that adds a 
development suite that does quick-switching like that~

ZwerOxotnik commented:
Do you know any tools which helps to make/update .md files with custom (js/xml etc.) data/GitHub data for repositories? Well, for example, it'd pretty nice to generate some data for Factorio settings and .md files at the same time.

ElectronicsArchiver commented:
You mean just including a piece of code 
as a code block in a markdown file?

If so, yes I saw a project doing that,
though I'm not sure how well it works.

https://github.com/zakhenry/embedme

In general I'd just go for writing
a mini tool doing something like
that, usually with **[Deno]**.
( Deno = NodeJS successor )

[Deno]: https://deno.land/

ZwerOxotnik commented:
> [embedme](https://github.com/zakhenry/embedme)

Well, it won't take care of links and won't generate .md content by some "rules" using some data, but it seems useful.

ElectronicsArchiver commented:
bump

ZwerOxotnik commented:
Perhaps, I'll check in ~23 hours.
The topic about "source" folder surprises me even more. Some devs name it as "Sourses".

ElectronicsArchiver commented:
@ZwerOxotnik You didn't find any time it seems?

ZwerOxotnik commented:
@ElectronicsArchiver, not really, I had been messing around with various stuff. Eventually, I'll check the zip script on several OS.

ElectronicsArchiver commented:
I see~

ZwerOxotnik commented:
There are several issues with https://github.com/ElectronicsArchiver/factorio-WhereIsMyBody/blob/f86a654a242309ccf513040950aad7bf8fd354dd/.scripts/zip_mod.sh.

*   It packages everything into `Source` folder, Factorio doesn't recognize such structure for mods (https://wiki.factorio.com/Tutorial:Mod_structure)
*   Why the script clears the console? (I'm not sure how it'll affect logs of GitHub actions, although it shouldn't be issue for GitHub, however it'll be less convenient to combine such scripts with auto tasks/scripts/extensions of Visual Studio Code etc., because their output won't be always obvious. Especially with linting)

ElectronicsArchiver commented:
@ZwerOxotnik 

> It packages everything into Source folder, Factorio doesn't recognize such structure for mods

> my-armor-mod_0.3.6.zip
|- aFolderName/
  |-- control.lua
  |-- data.lua
  |-- info.json
  |-- thumbnail.png
  
It does though? The source folder is just the aFolderName folder?

ZwerOxotnik commented:
I tested it on Factorio 1.1.61 and it works, however if someone going to unpack the .zip file, then Factorio will "mark" the mod as invalid since it doesn't follow its name and it may overwrite etc. a folder with the same name. (though it's weird to have the same folder name for all mods...)  
(For some reason, I get notifications here only via my email)

ElectronicsArchiver commented:
@ZwerOxotnik You might have to adjust your watch settings for this repository.

Can you just tell me what I should change for it to work.
